code-gen/src/main/scala/scalapb/validate/compiler/Rule.scala

```diff
@@ -50,6 +52,17 @@ case class Rule(
   def withImport(name: String) = copy(imports = imports :+ name)
 
   def withPreamble(lines: String*) = copy(preamble = preamble ++ lines)
+
+  implicit class FieldDescriptorPimp(descriptor: FieldDescriptor) {
```

Comment:
I'd prefer this as a top-level private function in the `Rule` object, instead of an implicit conversion here (i.e. `getFullNameWithPackage(descriptor): String`). BTW, in ScalaPB we're now using the word `Extension` instead of `Pimp` to avoid the negative connotation.